CMakeLists.txt

```diff
@@ -23,6 +23,10 @@ project(ecdaa
 
 add_compile_options(-Werror -Wall -Wextra -std=c99)
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        SET(CMAKE_C_FLAGS "-g -O0 -fprofile-arcs -ftest-coverage")
```

Comment:
Use `add_compile_options(...)` like line 24, rather than `SET(CMAKE_C_FLAGS ...`.

Reply:
Ok, reason 1 makes sense.  Reason 2 doesn't. I think the semantics are the same.  Each subdirectory gets it's own copy of `CMAKE_C_FLAGS` initialized to the value from its parent directory. (Could be wrong on this.)

But reason 1 wins out regardless.  And yes, I'd switch both to `SET` to be consistent.

In that case you want 
```
SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0 -fprofile-arcs -ftest-coverage")
```

Otherwise, you're overriding the default build flags and any that the user specified on the command line (see http://voices.canonical.com/jussi.pakkanen/2013/03/26/a-list-of-common-cmake-antipatterns/).  `-g` is implied by the Debug variant, so no need to repeated that.

The whole construct could be shortened from
```
if (CMAKE_BUILD_TYPE STREQUAL "Debug")
    SET(CMAKE_C_FLAGS ...)
endif()
```

to 

```
SET(CMAKE_C_FLAGS_DEBUG ...)
```

as well.

Reply:
As for the differences between the two options, I was going off of https://stackoverflow.com/questions/39501481/difference-between-add-compile-options-and-setcmake-cxx-flags

Yes, good call, I was clobbering the regular flags, good catch.

As for the suffix'd flags variable: that's cool, I didn't know about that!

tl;dr Agree. I've updated to use only `SET(CMAKE_C_FLAGS...`, with the `_DEBUGWITHCOVERAGE` variant used to add the gcov stuff.